app/main.py

```diff
@@ -1,2 +1,86 @@
+from typing import Generator
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self._size = 8
+        self._threshold = 0.3
+        self._table: list[list] = [[] for _ in range(self._size)]
+        self._resize_const = 2
+
+    def print_table(self) -> None:
+        print(self._table)
+
+    def __getitem__(self, key: any) -> all:
+        hash_key = hash(key)
+        table_index = hash_key % self._size
+
+        for _ in range(self._size):
+            if self._table[table_index] and self._table[table_index][1] == key:
+                return self._table[table_index][0]
+            else:
+                table_index = next(self._gen_next_slot(table_index))
+                if (self._table[table_index]
+                        and self._table[table_index][1] == key):
+                    return self._table[table_index][0]
+
+        raise KeyError
+
+    def __setitem__(self, key: any, value: any) -> None:
+        if isinstance(key, (list, set, dict)):
+            raise KeyError
+
+        if len(self) >= self._threshold * self._size:
+            self._resize_table()
+
+        hash_key = hash(key)
+        table_index = hash_key % self._size
+
+        for _ in range(self._size):
+            if not self._table[table_index]:
+                self._table[table_index] = [value, key]
+                return
+            else:
+                if self._table[table_index][1] == key:
+                    self._table[table_index] = [value, key]
+                    return
+
+                table_index = next(self._gen_next_slot(table_index))
+
+        print("NO EMPTY SLOTS")
```

Comment:
Instead of printing 'NO EMPTY SLOTS', it would be better to raise an exception or handle this case more gracefully. Consider raising an exception like `RuntimeError('No empty slots available')` to indicate that the dictionary is full and cannot accommodate more entries.